otherlibs/systhreads/st_stubs.c

```diff
@@ -13,6 +13,10 @@
 /*                                                                        */
 /**************************************************************************/
 
+#if defined(__sun) && !defined(_POSIX_PTHREAD_SEMANTICS)
+#define _POSIX_PTHREAD_SEMANTICS
+#endif
```

Comment:
As mentioned [here](https://github.com/ocaml/ocaml/pull/10063#issuecomment-755153727), I would have a preference for removing this change if the PTHREAD_CFLAGS change suffices to achieve this. (I prefer to have system-specific code in the configuration only.)

Reply:
Done.

Reply:
Thanks!